### PR TITLE
[LLVM10] Fixing compilation warnings for Reco2

### PR DESCRIPTION
**Pull request by smuzaffar**
#### PR description:

LLVM 10 integration in CLANG IBs shows new compilation warnings. This PR fixes few of these from reconstruction packages
- Avoid creating loop variable copy
- fix implicit conversion warnings

#### PR validation:

Local compilation for CLANG and normal IBs show no warnings.

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

File: RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromRawProducer.cc [hunk 1]
```diff
@@ -424,7 +424,7 @@ void ClusterFiller::fill(StripClusterizerAlgorithm::output_t::TSFastFiller& reco
             //rawAlgos_->suppressor->suppress( digis, zsdigis);
             uint16_t firstAPV = ipair * 2;
             rawAlgos.suppressVirginRawData(id, firstAPV, digis, zsdigis);
-            for (const auto digi : zsdigis) {
+            for (const auto& digi : zsdigis) {
```
Comment (slava77) [inline RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromRawProducer.cc:427]: this is coupled to the change made in #29568 and will not be necessary after that PR is merged.
please revert

File: RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc [hunk 2]
```diff
@@ -445,7 +445,7 @@ void MuonIdProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
 
   // muons first - no cleaning, take as is.
   if (muonCollectionHandle_.isValid()) {
-    for (const auto muon : *muonCollectionHandle_) {
+    for (const auto& muon : *muonCollectionHandle_) {
```
Comment (slava77) [inline RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc:448]: this is a duplicate of https://github.com/cms-sw/cmssw/pull/29568/files#diff-89dc8f122700a0c3b857022fe730fd42R448
please revert or merge the two PRs

File: RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromRawProducer.cc (hunk 1 above)
Comment (smuzaffar) [reply RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromRawProducer.cc:427]: thanks @slava77 , I will merge the two PRs

File: RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc (hunk 2 above)
Comment (smuzaffar) [reply RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc:448]: #29568 is now merged in this PR.

**Comment by smuzaffar** [comment]
please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/29572	69fe774 
- technical; code changes are in line with the PR description and the follow up review
- jenkins tests pass and comparisons with the baseline show no (relevant) differences

**Comment by civanch** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**